### PR TITLE
add staking message and API

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

File: proto/types/action.proto [hunk 1]
```diff
@@ -35,11 +37,32 @@ message PutPollResult {
 }
 
 message Execution {
-  string amount  = 1;
+  string amount = 1;
   string contract = 2;
   bytes data = 3;
 }
 
+message Pygg {
```
Comment (yutongp) [inline proto/types/action.proto]: Pygg -> `StackingBucket`

File: proto/types/action.proto [hunk 2]
```diff
@@ -35,11 +37,32 @@ message PutPollResult {
...
+  bytes canName = 1;
```
Comment (yutongp) [inline proto/types/action.proto]: canName -> votedCandidate

Comment (yutongp) [reply proto/types/action.proto]: string type?

File: proto/types/action.proto [hunk 3]
```diff
@@ -35,11 +37,32 @@ message PutPollResult {
...
+  bytes stakedAmount = 2;
```
Comment (yutongp) [inline proto/types/action.proto]: string time?

File: proto/types/action.proto [hunk 4]
```diff
@@ -42,25 +42,31 @@ message Execution {
...
   google.protobuf.Timestamp unstakeStartTime = 6;
   bool nonDecay = 7;
-  string pyggOwner = 8;
+  string owner = 8;
 }
 
-message Staking {
+message StakeCreation {
+  bytes canName = 1;
+  string stakedAmount = 2;
```
Comment (dustinxie) [inline proto/types/action.proto:58]: change to 'string' to be consistent with Transfer/Execution

**Review by yutongp — APPROVED** [review]

File: proto/types/action.proto [hunk 5]
```diff
@@ -35,11 +37,43 @@ message PutPollResult {
...
+
+message StakeCreate {
+  string candidateName = 1;
+  string stakedAmount = 2;
+  uint32 stakedDuration = 3;
+  bool nonDecay = 4;
+  bytes payload = 5;
+}
+
+// add, unstake, or withdraw
```
Comment (koseoyoung) [inline proto/types/action.proto:64]: need amount field to add amount

**Pull request merged**